### PR TITLE
[move-prover] Updating the Boogie translation to reflect new memory model

### PR DESCRIPTION
**Pull request by shazqadeer**
## Motivation

This PR moves Boogie translation to the new memory model.  In addition to bug fixes in the core transforms and analyses, the bulk of the changes are in bytecode_translator.rs.  The interaction between the new memory model and explicit PackRef/UnpackRef and resource invariants is not fully sorted out.  Specifically, there are several discrepancies in the output for the test case tests/sources/functional/mut_ref_accross_modules.exp.  I have noted down these discrepancies for further investigation.
 
### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Existing tests

### CODE REVIEW

File: language/move-prover/tests/sources/functional/aborts-if.exp [hunk 1]
```diff
@@ -8,8 +8,6 @@ error:  A postcondition might not hold on this return path.
     │
     =     at tests/sources/functional/aborts-if.move:32:5: abort2_incorrect (entry)
     =     at tests/sources/functional/aborts-if.move:32:5: abort2_incorrect (exit)
-    =         _x = <redacted>,
```
Comment (wrwg) [inline language/move-prover/tests/sources/functional/aborts-if.exp:11]: Looks like all debug info is gone here. It's likely OK we submit this, but we need to get this back ASAP, given that its needed for folks working on the specs.

File: language/move-prover/tests/sources/functional/mut_ref_accross_modules.move [hunk 2]
```diff
@@ -76,22 +76,22 @@ module TestMutRefs {
     }
 
     // For a public function, the data invariant for a mut ref is expected to hold.
-    public fun data_invariant(x: &mut T) {
+    public fun data_invariant(x: &mut T) { // Shaz: This function verifies because resource invariant is assumed on the inout value modeling x
```
Comment (wrwg) [inline language/move-prover/tests/sources/functional/mut_ref_accross_modules.move:79]: Is this a comment or an issue we need to fix? If it is an issue please clearly mark a TODO here and state it needs to be fixed. If not I'm wondering what the comment is about. if its a clarification, please remove redundant information ("Shaz") and put the comment before the function. (Which also says something similar?) Same for other cases.

I remember you said that where issues with some tests but I can't find them documented in the PR.

Comment (shazqadeer) [reply language/move-prover/tests/sources/functional/mut_ref_accross_modules.move]: All the issues were in these "Shaz:" comments in this one test.  I have improved the comments to explain the issue more precisely.

File: language/move-prover/tests/sources/functional/aborts-if.exp (hunk 1 above)
Comment (shazqadeer) [reply language/move-prover/tests/sources/functional/aborts-if.exp:11]: Agreed.  Debug info and the other test you commented on are the two major outstanding issues once this PR lands.

**Pull request closed without merging**